agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/Configuration.java

```diff
@@ -62,20 +64,46 @@ private static boolean isEmpty(String str) {
     }
 
     public enum MatchType {
-        // Moshi JSON builder do not allow case insensitive mapping
-        strict, regexp
+        @Json(name = "strict")
+        STRICT {
```

Comment:
it's not clear to me why you need to define MatchType in both places.. one here and calling MetricFilter's constants.. why not just keep one enum here?